src/Microsoft.CodeQuality.Analyzers/Core/Maintainability/ReviewUnusedParameters.cs

```diff
@@ -195,6 +195,47 @@ public void AnalyzeOperation(OperationAnalysisContext context)
 
             public void OperationBlockEndAction(OperationBlockAnalysisContext context)
             {
+                // Check to see if the method just throws a NotImplementedException. If it does, we shouldn't warn
+                // about parameters. Note that VB method bodies with 1 action have 3 operations.
+                // The first is the actual operation, the second is a label statement, and the third is a return
+                // statement. The last two are implicit in these scenarios.
+                if (context.OperationBlocks.Length == 1)
+                {
+                    bool IsSingleStatementBody(IBlockOperation body)
+                    {
+                        return body.Operations.Length == 1 ||
+                            (body.Operations.Length == 3 && body.Syntax.Language == LanguageNames.VisualBasic &&
+                             body.Operations[1] is ILabeledOperation labeledOp && labeledOp.IsImplicit &&
+                             body.Operations[2] is IReturnOperation returnOp && returnOp.IsImplicit);
+                    }
+
+                    var methodBlock = (IBlockOperation)context.OperationBlocks[0];
+                    if (IsSingleStatementBody(methodBlock))
+                    {
+                        var innerOperation = methodBlock.Operations.First();
+
+                        // Because of https://github.com/dotnet/roslyn/issues/23152, there can be an expression-statement
+                        // wrapping expression-bodied throw operations. Compensate by unwrapping if necessary.
+                        if (innerOperation.Kind == OperationKind.ExpressionStatement &&
+                            innerOperation is IExpressionStatementOperation exprStatement)
+                        {
+                            innerOperation = exprStatement.Operation;
+                        }
+
+                        if (innerOperation.Kind == OperationKind.Throw &&
+                            innerOperation is IThrowOperation throwOperation &&
+                            throwOperation.Exception.Kind == OperationKind.ObjectCreation &&
+                            throwOperation.Exception is IObjectCreationOperation createdException)
+                        {
+                            var notImplementedType = WellKnownTypes.NotImplementedException(context.Compilation);
```

Comment:
:bulb: This should include `NotSupportedException` as well.

Reply:
Added.